node/flatpak_node_generator/providers/yarn.py

```diff
@@ -37,88 +39,72 @@ def is_git_version(version: str) -> bool:
 
-        version: Optional[str] = None
-        resolved: Optional[str] = None
-        integrity: Optional[Integrity] = None
-
-        section_indent = 0
-
-        line = None
-        for line in section[1:]:
-            indent = 0
-            while line[indent].isspace():
-                indent += 1
-
-            assert indent, line
-            if not section_indent:
-                section_indent = indent
-            elif indent > section_indent:
-                # Inside some nested section.
-                continue
-
-            line = line.strip()
-
-            if line.startswith('"'):
-                # XXX: assuming no spaces in the quoted region!
-                key, value = line.split(' ', 1)
-                line = f'{self.unquote(key)} {value}'
-
-            if line.startswith('version'):
-                version = self.unquote(line.split(' ', 1)[1])
-            elif line.startswith('resolved'):
-                resolved = self.unquote(line.split(' ', 1)[1])
-            elif line.startswith('integrity'):
-                _, values_str = line.split(' ', 1)
-                values = self.unquote(values_str).split(' ')
-                integrity = Integrity.parse(values[0])
-
-        assert version, section
-
         source: PackageSource
         if self._LOCAL_PKG_RE.match(version_constraint):
             source = LocalSource(path=self._LOCAL_PKG_RE.sub('', version_constraint))
         else:
-            assert resolved, section
-            if self.is_git_version(resolved):
-                source = self.parse_git_source(version=resolved)
+            if self.is_git_version(entry['resolved']):
+                source = self.parse_git_source(version=entry['resolved'])
             else:
-                source = ResolvedSource(resolved=resolved, integrity=integrity)
+                if 'integrity' in entry:
+                    integrity = Integrity.parse(entry['integrity'])
+                else:
+                    integrity = None
+                source = ResolvedSource(resolved=entry['resolved'], integrity=integrity)
 
-        return Package(name=name, version=version, source=source, lockfile=lockfile)
+        return Package(
+            name=name, version=entry['version'], source=source, lockfile=lockfile
+        )
 
     def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
-        section: List[str] = []
-
-        with open(lockfile) as fp:
-            for line in map(str.rstrip, fp):
-                if not line.strip() or line.strip().startswith('#'):
-                    continue
-
-                if not line[0].isspace():
-                    if section:
-                        yield self.parse_package_section(lockfile, section)
-                        section = []
-
-                section.append(line)
-
-        if section:
-            yield self.parse_package_section(lockfile, section)
+        for name, package in self.parse_lockfile_v1(lockfile).items():
```

Comment:
```suggestion
        for name_line, package in self.parse_lockfile_v1(lockfile).items():
```

since these still aren't *actually* names, each line might have >1 name on it separated by `,`s, e.g. from the electron lock file:

```
mimic-response@^1.0.0, mimic-response@^1.0.1:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/mimic-response/-/mimic-response-1.0.1.tgz#4923538878eef42063cb8a3e3b0798781487ab1b"
  integrity sha512-j5EctnkH7amfV/q5Hgmoal1g2QHFJRraOtmx0JpIqkxhBhI/lJSl1nMpQ45hVarwNETOoWEimndZ4QK0RHxuxQ==
```